src/core.rs

```diff
@@ -303,6 +303,30 @@ impl Core {
         self.call_edit("scroll", Some(json!([start, end])))
     }
 
+    pub fn resize(&mut self, height: u16) -> Result<()> {
+        let window = self.get_view().unwrap().get_window();
+
+        // we have no way to know if the screen was resized from the top or the bottom, so we
+        // balance the change between both end.
+        // basically we want:
+        //  1) new_height = new_end - new_start
+        //  2) new_end - old_end = new_start - old_start
+        // which leads to:
+        //  1) new_end = (new_height + old_start + old_end) / 2
+        //  2) new_start = (old_start + old_end - new_height) / 2
+        //
+        // we just to handle corner cases since we're working with unsigned integer.
+
+        let mut new_start: u64;
+        if height as u64 > window.0 + window.1 {
+            new_start = 0;
+        } else {
+            new_start = window.0 + window.1 - height as u64 / 2;
```

Comment:
you need parens here, assuming the comment above is correct.

should be: `(window.0 + window.1 - height) / 2`